linode/databases/framework_datasource_schema.go

```diff
@@ -0,0 +1,117 @@
+package databases
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/terraform-provider-linode/linode/helper/frameworkfilter"
+)
+
+var filterConfig = frameworkfilter.Config{
+	"engine":  {TypeFunc: frameworkfilter.FilterTypeString},
+	"region":  {TypeFunc: frameworkfilter.FilterTypeString},
+	"status":  {TypeFunc: frameworkfilter.FilterTypeString},
+	"type":    {TypeFunc: frameworkfilter.FilterTypeString},
+	"version": {TypeFunc: frameworkfilter.FilterTypeString},
+
+	"allow_list":     {TypeFunc: frameworkfilter.FilterTypeString},
+	"cluster_size":   {TypeFunc: frameworkfilter.FilterTypeInt},
+	"created":        {TypeFunc: frameworkfilter.FilterTypeString},
+	"encrypted":      {TypeFunc: frameworkfilter.FilterTypeBool},
+	"host_primary":   {TypeFunc: frameworkfilter.FilterTypeString},
+	"host_secondary": {TypeFunc: frameworkfilter.FilterTypeString},
+	"id":             {TypeFunc: frameworkfilter.FilterTypeInt},
+	"instance_uri":   {TypeFunc: frameworkfilter.FilterTypeString},
+	"updated":        {TypeFunc: frameworkfilter.FilterTypeString},
+	"label":          {TypeFunc: frameworkfilter.FilterTypeString},
+}
```

Comment:
`label`, `region`, `status`, `type`, `version` are API filterable, right?

Reply:
Looks like these fields are documented as filterable, but the API returns `[X-Filter] Could not apply filter` if any of these fields are included in the constructed filter. I'm not sure if this is a docs issue or an API issue, but I'm leaning towards it being an API issue.

Reply:
Just confirmed that this is an API issue and was able to pinpoint the cause of the bug. I've reported it to the API team internally 👍

Reply:
Thanks for the insightful investigation!